engines/bops_api/app/services/bops_api/application/creation_service.rb

```diff
@@ -144,7 +147,9 @@ def possibly_immune?(planning_application)
       def process_immunity_details(planning_application)
         ActiveRecord::Base.transaction do
           immunity_detail = ImmunityDetail.new(planning_application: planning_application)
-          immunity_detail.end_date = planning_application
+
+          immunity_detail.end_date = planning_application.params_v2.dig("data", "proposal", "date", "completion")
+          immunity_detail.end_date ||= planning_application
```

Comment:
Why are we setting the `end_date` to the planning application if the above line is nil?

Reply:
We're not. We're setting it to the applicant's response from the proposal details, i.e., the previous behaviour, in case the `date` field is blank but the question exists — which seems to be the case in the test data.